### ISSUE TITLE
Référentiel: problème de rafraichissement de la table

### ISSUE THREAD

**Issue by camillecarrier** [alexandresoro/ouca#1219]
Pour tous les types d'entité du référentiel

# Problème 1
- Créer une commune avec le nom "a"
- Trier les communes par nom par ordre alphabétique du nom
-> la commune a n'apparait pas.

Par contre si on scroll jusqu'en bas de la table elle apparait.

Même problème quand on modifie la commune.

# Problème 2
- Modifier une commune
- Scroller tout en bas de la table jusqu'à la commune
- Cliquer sur modifier
- Les anciennes valeurs apparaissent dans le formulaire

# Problème 3
- Créer une classe espèce "test"
- Filtrer les classes espèces par "test"
-> la classe créée n'apparait pas

**Comment by alexandresoro** [comment]
ah, looks like a regression I introduced with the migration to infinite queries.
Looks like `refetch` is not the way to go with infinite queries, and should be `queryClient.restQueries(...)` instead 🤯

**Comment by alexandresoro** [comment]
So Problem 3 is probably linked to a cache that was not properly invalidated, and fixed with #1261

**Comment by alexandresoro** [comment]
Problem 1 is most probably linked to the fact that default sorting is per id? to be confirmed. If so, let's see how we can filter by town name by default. And make sure it is in sync with UI. Possibly the way to go is to set a default sort value directly from the UI, so that we make sure that we get what we look for.

Problem 2 to be investigated in how it can be reproduced. To be checked after #1251 so that we make sure that this is fixed.

**Comment by alexandresoro** [comment]
On second thought:

**Problem 1** 

Not sure to understand what was wrong, but cache issues have been fixed by #1261. I tried to recreate a town "a" and by default it appears at the end, since there's no default sorting. Doing the same with sorting by name set beforehand does display "a" first, and correctly replaces it on update. 
If the remaining point is about having a default sorting, I suggest that we track it in a separate ticket as we need to enumerate the most sensible default sorting per entity.

**Problem 2**

Looks like the same issue as #1220. I'll follow up there.

**Problem 3**

I strongly suspect that the table was filtered with "test" before in the scenario, and that we reused a cached resource. This has been fixed in #1261 .

@camillecarrier I have loaded a bunch of fixes on dev. Could you please check the issues raised here and confirm that this is ok now for P1 and P3.
Do you confirm that P2 is similar to #1220 too?
